app/scripts/metamask-controller.js

```diff
@@ -6341,12 +6341,6 @@ export default class MetamaskController extends EventEmitter {
           this.approvalController.addAndShowApprovalRequest.bind(
             this.approvalController,
           ),
-        startApprovalFlow: this.approvalController.startFlow.bind(
-          this.approvalController,
-        ),
-        endApprovalFlow: this.approvalController.endFlow.bind(
-          this.approvalController,
-        ),
```

Comment:
why are we removing these? They don't appear to have anything to do with the bug? Are they just obsolete now?

Reply:
they're unneeded because of the changes that this fix introduces

Reply:
Actually i'm wrong. Technically they were obsolete after the isAddFlow was added, i.e. you are right and they aren't directly related to this bug. I'll remove them separately

Reply:
I forgot I had already done this work weeks ago here https://github.com/MetaMask/metamask-extension/pull/28189

Reply:
this has been merged